Odata-docs/webapiauth/how-permissions-are-applied.md

```diff
@@ -0,0 +1,119 @@
+
+On each request, the library extracts from the model the permissions restrictions that should apply to the route being accessed and creates an authorization policy based on those permissions. Deeper down the request pipeline, the AspNetCore filter-based authorization system will call the OData authorization handler to verify whether the current user's permissions match the ones required by the policy.
+
+**Note**: If there are not permission restrictions defined for an some target (entity set/singleton/operation) in the model, then endpoints to that target will be authorized by default regardless of the user's permissions.
+
+## Example permission scopes
+
+For the following examples, let's assume that we are working with an OData model that has the following scopes defined:
+
+
+Permission scope name     | Where it's defined
+--------------------------|--------------------
+`Customers.Read`          | `ReadRestrictions` of `Customers` entity set
+`Customers.ReadByKey`     | `ReadByKeyRestrictions` of `Customers`
+`Customers.Insert`        | `InsertRestrictions` of `Customers`
+`Customers.Delete`        | `DeleteRestrictions` of `Customers`
+`Customers.Update`        | `UpdateRestrictions` of `Customers`
+`CustomerOrders.Read`     | `ReadRestrictions` of `NavigationRestrictions` of `Customers` on `Orders` property
+`CustomerOrders.ReadByKey` | `ReadByKeyRestriction` of `NavigationRestrictions` of `Customers` on `Orders` property
+`CustomerOrders.Insert` | `InsertRestrictions` of `NavigationRestrictions` of `Customers` on `Orders` property
+`CustomerOrders.Update` | `UpdateRestrictions` of `NavigatonRestrictions` of `Customers` on `Orders` property
+`CustomerOrders.Delete` | `UpdateRestrictions` of `NavigationRestrictions` of `Customers` on `Orders` property
+`Orders.Read`           | `ReadRestrictions` of `Orders` entity set
+`Orders.ReadByKey`      | `ReadByKeyRestrictions` of `Orders`
+`Orders.Update`         | `UpdateRestrictions` of `Orders`
+`Orders.Delete`         | `DeleteRestrictions` of `Orders`
+`Orders.Insert`         | `InsertRestrictions` of `Orders`
+`Order.CalculateTax`          | `OperationRestrictions` of `CalculateTax` bound function
+`UpdateTaxRate`        | `OperationRestrictions` of `UpdateTaxRate` unbound action
+`TopProduct.Read`      | `ReadRestrictions` of `TopProduct` singleton
+
+## CRUD operations on entity sets and singleton
+
+For CRUD operations on entity sets and singleton, the permissions of the corresponding insert/update/delete/read restrictions are applied.
+
+Endpoint                     | Required permission scopes
+-----------------------------|----------------------
+`GET Customers`              | `Customers.Read`
+`GET Customers(1)`           | `CustomersRead OR Customers.ReadByKey`
+`DELETE Customers/1`         | `Customers.Delete`
+`POST Customers`             | `Customers.Insert`
+`PUT Customers`              | `Customers.Update`
```

Comment:
Just to confirm, UpdateRestrictions apply to both PUT and PATCH operations, but we don't consider a scenario where you can create using a PUT operation. .....Right? This would make it ambiguous for `PUT Customers` because we wouldn't be sure whether to apply InsertRestrictions or to apply UpdateRestrictions

Reply:
Yeah currently UpdateRestrictions apply to `PUT` and `PATCH` (and `MERGE`). The library makes no attempt to apply `InsertRestrictions` to `PUT`.